### ISSUE TITLE
Alpine-based apache 2.4.37 does not start with mod_ssl

### ISSUE THREAD

**Issue by csandanov** [docker-library/httpd#115]
The latest version does not start with enabled mod_ssl:
```
httpd: Syntax error on line 159 of /usr/local/apache2/conf/httpd.conf: 
Cannot load modules/mod_ssl.so into server: 
Error relocating /usr/local/apache2/modules/mod_ssl.so: SSL_CTX_set_post_handshake_auth: symbol not found
```
Perhaps requires the latest libressl version, I see a lot of compatibility fixes have been made in 2.8.1 https://github.com/libressl-portable/portable/blob/master/ChangeLog

**Comment by sergey-zabolotny** [comment]
the same error

**Comment by laurenra7** [comment]
This can be replicated by building a container from the latest **httpd:alpine** image and following the instructions for **SSL/HTTPS** on the [Docker Hub page](https://hub.docker.com/_/httpd/).

`httpd: Syntax error on line 159 of /usr/local/apache2/conf/httpd.conf: Cannot load modules/mod_ssl.so into server: Error relocating /usr/local/apache2/modules/mod_ssl.so: SSL_CTX_set_post_handshake_auth: symbol not found`

**Comment by yosifkit** [comment]
Perhaps the bump to `alpine:3.8` in https://github.com/docker-library/httpd/pull/114 will help.  You can build the new [Dockerfile](https://github.com/docker-library/httpd/blob/b2e7d2868e2f92660469ac66187f8f83fe449c65/2.4/alpine/Dockerfile) or they should be on the Docker Hub once https://github.com/docker-library/official-images/pull/4994 is merged and built.

**Comment by csandanov** [comment]
@yosifkit I tried running with 3.8 and edge, the same thing. The latest libressl in alpine repo is 2.7.4 atm.

**Comment by designbyjr** [comment]
I can also confirm that this also affects php:7.1.20-apache. Bump on upgrade, I've a VPS waiting here for SSL.

**Comment by tianon** [comment]
`php:7.1.20-apache` is completely unrelated to this image -- it installs Apache from Debian where this one builds it from source.

**Comment by badgersow** [comment]
We are also affected by this problem and will temporarily use `2.4.35-alpine` version

**Comment by radiocity** [comment]
@badgersow, it works after `apk add --no-cache 'openssl<1.1.0'` in my case

**Comment by ptaunton** [comment]
Current usable image (2.4.35) is being flagged as having a vulnerability in libxml2 so this needs to be fixed as soon as possible.

**Comment by yosifkit** [comment]
> [stretch] - libxml2 <postponed> (Minor issue; wait for upstream fix for upstream bug 794914)
>
> \- https://security-tracker.debian.org/tracker/CVE-2017-18258

The CVE is irrelevant since even if we rebuilt the image the CVE would not be fixed anyway.

**Comment by mduft** [comment]
Same issue here. Since I'm building a patched apache anyway myself, I (for now) reverted back to alpine 3.7 and it works.

**Comment by AliceWonderMiscreations** [comment]
Could this be same issue referenced here?

http://openbsd-archive.7691.n7.nabble.com/Apache-2-4-37-SSL-CTX-set-post-handshake-auth-error-td354171.html#a354452

**Comment by pesochek** [comment]
Ended up switching to nginx in my docker-compose projects..

**Comment by bronger** [comment]
> Ended up switching to nginx in my docker-compose projects..

The non-Alpine httpd works well.

**Comment by pesochek** [comment]
> > Ended up switching to nginx in my docker-compose projects..
> 
> The non-Alpine httpd works well.

There's an Alpine based nginx though and it works fine with SSL. That's what I meant.

**Comment by bronger** [comment]
I indended my comment for those people for whom it’s easier to drop “:alpine” rather than writing new config for another software product.  ;-)

**Comment by tianon** [comment]
Looking into this more, it's actually kind of odd that we don't get any errors during compile of Apache -- it's trying to use `SSL_CTX_set_post_handshake_auth`, which as far as I can tell was added in OpenSSL 1.1.1 and does not (yet?) exist in LibreSSL, which is what we have to use in Alpine due to this fun dependency mess:

```console
ERROR: unsatisfiable constraints:
  libressl-dev-2.7.4-r0:
    conflicts:
               openssl-dev-1.0.2q-r0[pc:libcrypto=2.7.4]
               openssl-dev-1.0.2q-r0[pc:libssl=2.7.4]
               openssl-dev-1.0.2q-r0[pc:openssl=2.7.4]
    satisfies:
               apr-util-dev-1.6.1-r3[libressl-dev]
               mariadb-dev-10.2.19-r1[libressl-dev]
               postgresql-dev-10.5-r0[libressl-dev]
               openldap-dev-2.4.46-r0[libressl-dev]
               mariadb-connector-c-dev-3.0.4-r1[libressl-dev]
  openssl-dev-1.0.2q-r0:
    conflicts:
               libressl-dev-2.7.4-r0[pc:libcrypto=1.0.2q]
               libressl-dev-2.7.4-r0[pc:libssl=1.0.2q]
               libressl-dev-2.7.4-r0[pc:openssl=1.0.2q]
  .build-deps-0:
    masked in: cache
    satisfies: world[.build-deps]
```

Also, more interestingly, I can reproduce with `apache2-ssl` from Alpine:

```console
FROM alpine:3.8
RUN apk add --no-cache apache2-ssl
CMD ["httpd", "-DFOREGROUND"]
```

Running the result of building that gives the following:

```console
httpd: Syntax error on line 480 of /etc/apache2/httpd.conf: Syntax error on line 13 of /etc/apache2/conf.d/ssl.conf: Cannot load modules/mod_ssl.so into server: Error relocating /var/www/modules/mod_ssl.so: SSL_CTX_set_post_handshake_auth: symbol not found
```

Any ideas, @ncopa?  Is this a bug in Apache that it isn't detecting LibreSSL properly?  Or is it a bug in LibreSSL that it doesn't appear to include this function that Apache is trying to use?

**Comment by csandanov** [comment]
Alpine Linux 3.9 has been released where they migrated back from LibreSSL to OpenSSL. I think once the image rebased to 3.9 this issue should be fixed.

**Comment by ovk** [comment]
Appears to be fixed in Alpine 3.9 based image.
